server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -279,6 +279,10 @@ protected void doRun() {
         });
     }
 
+    static boolean isDedicatedVotingOnlyNode(DiscoveryNode node) {
+        return node.isMasterNode() && node.isDataNode() == false &&
+            node.getRoles().stream().anyMatch(role -> role.roleName().equals("voting_only"));
```

Comment:
Can we establish that this is the same string as used in `VotingOnlyNodePlugin#VOTING_ONLY_NODE_ROLE` by extracting the string as a field and adding an assertion in `VotingOnlyNodePlugin` saying that they match?

Reply:
This is ensured by the test that was added. It fails when these strings are not matching (as we are then illegally accessing the repo on the voting-only node)? Do you find this too implicit?